client/a8c-for-agencies/sections/referrals/referrals-list/index.tsx

```diff
@@ -108,11 +118,60 @@ export default function ReferralList( { referrals, dataViewsState, setDataViewsS
 		[ dataViewsState.selectedItem, isDesktop, openSitePreviewPane, translate ]
 	);
 
+	useEffect( () => {
+		// If the user clicks on a row, open the preview pane by triggering the View details button click.
+		const handleRowClick = ( event: Event ) => {
+			const target = event.target as HTMLElement;
+			const row = target.closest(
+				'.dataviews-view-table__row, li:has(.dataviews-view-list__item)'
+			);
+
+			if ( row ) {
+				const isButtonOrLink = target.closest( 'button, a' );
+				if ( ! isButtonOrLink ) {
+					const button = row.querySelector( '.view-details-button' ) as HTMLButtonElement;
+					if ( button ) {
+						button.click();
+					}
+				}
+			}
+		};
+
+		const rowsContainer = document.querySelector( '.dataviews-view-table, .dataviews-view-list' );
+
+		if ( rowsContainer ) {
+			rowsContainer.addEventListener( 'click', handleRowClick as EventListener );
+		}
+
+		// We need to trigger the click event on the View details button when the selected item changes to ensure highlighted row is correct.
+		if (
+			rowsContainer?.classList.contains( 'dataviews-view-list' ) &&
+			dataViewsState?.selectedItem?.client_id
+		) {
+			const trigger: HTMLButtonElement | null = rowsContainer.querySelector(
+				`li:not(.is-selected) .view-details-button[data-client-id='${ dataViewsState?.selectedItem?.client_id }']`
+			);
+			if ( trigger ) {
+				trigger.click();
+			}
+		}
+
+		return () => {
+			if ( rowsContainer ) {
+				rowsContainer.removeEventListener( 'click', handleRowClick as EventListener );
+			}
+		};
+	}, [ dataViewsState ] );
```

Comment:
I couldn't find an elegant way to do this, but I found the WPCOM implementation does this hack, and I followed the implementation. https://github.com/Automattic/wp-calypso/blob/trunk/client/sites-dashboard-v2/sites-dataviews/index.tsx#L78